### ISSUE TITLE
Linux port

### ISSUE THREAD

**Issue by auvipy** [ReactiveCocoa/ReactiveCocoa#2600]

**Comment by NachoSoto** [comment]
I imagine there's not much work to be done, since the Cocoa/Mac additions are only added for iOS/Mac platforms.

**Comment by natestedman** [comment]
I got [somewhere](https://github.com/natestedman/ReactiveCocoa/tree/linux) on this.

The main things are:

- `swift build` needs to be supported, since there's no Xcode (https://github.com/ReactiveCocoa/ReactiveCocoa/pull/2599).
- `Result` needs to be updated for Linux - I did that, but it's an `#if os`-feature disabling based implementation. I believe that RAC doesn't use any of the disabled features, though!
- Foundation need to be disabled - some will be able to be added later, as the Foundation implementation in Swift advances.
- `OS...` functions don't exist - I replaced the locks on non-Apple platforms with `NSLock` for now - it's already implemented in the Swift version of Foundation. But then I stopped, because...
- `libdispatch` hasn't been ported yet, so none of those symbols work.

So basically, currently this can't be done without a *huge* amount of work that would be nullified once `libdispatch` is ported - so I stopped at that point. Once that does happen, it *should* work. There might be additional errors that are currently invisible because of all of the `dispatch_` ones.

That's for the Swift API only - haven't done anything with the Objective-C API as `swift build` doesn't support it currently.
